src/overlays/actors/ovl_Bg_Spot16_Bombstone/z_bg_spot16_bombstone.c

```diff
@@ -21,49 +111,445 @@ const ActorInit Bg_Spot16_Bombstone_InitVars = {
+    EnBombf* playerHeldActor;
+    Player* player = PLAYER;
+    EnBombf* currentBomb;
+
+    if (sTimer > 0) {
+        sTimer--;
+    }
+
+    if (sPlayerBomb != NULL) {
+        if (sPlayerBomb->actor.update == NULL) {
+            sPlayerBomb = NULL;
+        } else if (sTimer <= 0 && sPlayerBomb->actor.posRot.pos.y < 1400.0f &&
+                   Math3D_Dist1DSq(sPlayerBomb->actor.posRot.pos.x + 1579.0f,
+                                   sPlayerBomb->actor.posRot.pos.z + 790.0f) < SQ(400.0f) &&
+                   sPlayerBomb->actor.params == 0) {
+            currentBomb = sPlayerBomb;
+            if (currentBomb->timer > 0) {
+                sTimer = currentBomb->timer + 20;
+                func_800800F8(globalCtx, 0x1054, sTimer, NULL, 0);
+            }
+        }
+    } else if (player->stateFlags1 & 0x800) {
+        playerHeldActor = player->heldActor;
+        if (playerHeldActor != NULL && playerHeldActor->actor.type == ACTORTYPE_EXPLOSIVES &&
+            playerHeldActor->actor.id == ACTOR_EN_BOMBF) {
+            sPlayerBomb = playerHeldActor;
+        }
+    }
+}
+
+void func_808B5934(BgSpot16Bombstone* this) {
+    this->actor.draw = BgSpot16Bombstone_Draw;
+    this->actionFunc = func_808B5950;
+}
+
+void func_808B5950(BgSpot16Bombstone* this, GlobalContext* globalCtx) {
+    u8 acFlags;
+
+    func_808B56BC(this, globalCtx);
+    func_808B57E0(this, globalCtx);
+
+    if (globalCtx) {}
+
+    acFlags = this->colliderCylinder.base.acFlags;
+
+    if (acFlags & 2) {
+        this->colliderCylinder.base.acFlags = acFlags & ~2;
+
+        func_808B561C(this, globalCtx);
+
+        func_800800F8(globalCtx, 0x1054, 0x32, NULL, 0);
+
+        Flags_SetSwitch(globalCtx, this->switchFlag);
+        gSaveContext.eventChkInf[2] |= 8;
+
+        func_808B5A78(this);
+    } else {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colliderCylinder);
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->colliderJntSph);
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colliderJntSph);
+    }
+
+    if (mREG(64) == 1) {
+        func_808B561C(this, globalCtx);
+        mREG(64) = -10;
+    } else if (mREG(64) < 0) {
+        mREG(64)++;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot16_Bombstone/func_808B4D04.s")
+void func_808B5A78(BgSpot16Bombstone* this) {
+    this->unk_154 = 0;
+    this->unk_158 = 0;
+    this->actor.draw = NULL;
+    this->actionFunc = func_808B5A94;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot16_Bombstone/func_808B4D9C.s")
+void func_808B5A94(BgSpot16Bombstone* this, GlobalContext* globalCtx) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot16_Bombstone/func_808B4E58.s")
+    func_808B5240(this, globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot16_Bombstone/BgSpot16Bombstone_Init.s")
+    if (this->unk_154 == 0x38) {
```

Comment:
`unk_154` is a timer so I think these values should be decimal
```suggestion
    if (this->unk_154 == 56) {
```